.github/actions/setup-go/action.yml

```diff
@@ -27,8 +27,9 @@ runs:
           ~/.cache/go-build
           ~/Library/Caches/go-build
           ~\AppData\Local\go-build
-        key: ${{ runner.os }}-go-${{ inputs.go-version }}-${{ hashFiles('**/go.sum') }}
+        key: ${{ runner.os }}-go-${{ inputs.go-version }}-${{ github.job }}-${{ hashFiles('**/go.sum') }}
```

Comment:
Interesting, so we were creating new cache for every job previously?

Reply:
Yes, since each job will have different things in the cache it makes sense to do that. Otherwise the first job that completes will create a cache (let's say the "check commits" job, which doesn't do much build cache wise). Then all other jobs will use that cache next time, but since it's almost empty, it doesn't speed up anything. And then the jobs won't update the cache because they had a full cache hit (when the full key matches).
Giving each job its own cache key means they can update their individual caches independently.